src/main/java/uk/gov/ons/census/casesvc/service/ReceiptService.java

```diff
@@ -20,41 +20,103 @@
   private final CaseService caseService;
   private final UacService uacService;
   private final EventLogger eventLogger;
+  private final FieldworkFollowupService fieldworkFollowupService;
 
-  public ReceiptService(CaseService caseService, UacService uacService, EventLogger eventLogger) {
+  public ReceiptService(
+      CaseService caseService,
+      UacService uacService,
+      EventLogger eventLogger,
+      FieldworkFollowupService fieldworkFollowupService) {
     this.caseService = caseService;
     this.uacService = uacService;
     this.eventLogger = eventLogger;
+    this.fieldworkFollowupService = fieldworkFollowupService;
   }
 
   public void processReceipt(ResponseManagementEvent receiptEvent) {
     ResponseDTO receiptPayload = receiptEvent.getPayload().getResponse();
     UacQidLink uacQidLink = uacService.findByQid(receiptPayload.getQuestionnaireId());
+
+    logEvent(receiptEvent, receiptPayload, uacQidLink);
+
+    // If the uacQidLink is already marked as a blank questionnaire, the uacQidLink is inactive and
+    // can't be changed
+    if (uacQidLink.isBlankQuestionnaireReceived()) {
+      return;
+    }
+
+    if (receiptPayload.isUnreceipt()) {
+      processUnreceiptingEvent(uacQidLink, receiptEvent);
+      return;
+    }
+
+    if (isCCSQuestionnaireType(uacQidLink.getQid())) {
```

Comment:
Again, if we treat CCS cases like normal cases (i.e. send 'em on to action and let action send the messages to Field) then this goes away